### PR TITLE
SPARK-13991 - Extend the enforcer plugin Maven version rule

### PR DESCRIPTION
**Pull request by jbonofre**
## What changes were proposed in this pull request?

Extend of the Maven version in the enforcer plugin configuration.
## How was this patch tested?

Full build with build/mvn -DskipTests clear package

### CODE REVIEW

**Comment by JoshRosen** [comment]
Why should we do this? It's one less variable / environment thing for us to have to worry about if we can rely on a fixed Maven version.

Alternatively, why not treat `${maven.version}` as a lower-bound?

**Comment by jbonofre** [comment]
Hi Josh,
it means that we force users to use a specific Maven version. For instance, I didn't have yet update to Maven 3.3.9, I'm still using 3.3.3.
If we force a version, it's OK for me (I will update on my machine), but I don't think it will be easy to maintain (updating for each new Maven release).
Maybe, we can at least specify [${maven.version},) (again, meaning that we really force the minimal Maven version).

**Comment by rxin** [comment]
@jbonofre thanks for the patch. However, I don't think things have really changed since your last patch https://github.com/apache/spark/pull/8598

The problem is that there are subtle differences from version to version, and the easiest way for us to ensure build reproducibility is to fix the version here.

**Comment by srowen** [comment]
`build/mvn` already downloads the latest for you, hence it's required. Just making it a lower bound is fine. I'd rather keep plugins and build tools up to date and enforce that.

**Comment by jbonofre** [comment]
As you explained in #8598, I understand the need to have Maven 3.3.x version (instead of Maven 3.2.x). It makes complete sense. The point in this PR is the fact that we "force" the latest Maven 3.3.x version, whereas the build works fine with another version in the Maven 3.3.x serie.
@srowen it's weird as build/mvn doesn't download the latest version of Maven on my box, it uses the one installed:

spark$ ./build/mvn -DskipTests clean package
Using `mvn` from path: /opt/apache-maven-3.3.3/bin/mvn
...
[WARNING] Rule 0: org.apache.maven.plugins.enforcer.RequireMavenVersion failed with message:
Detected Maven Version: 3.3.3 is not in the allowed range 3.3.9.

I think using enforcer [3.3,) rule for Maven version would be more "flexible".

Let me know what it makes sense, else I will close this PR.

Thanks !

File: pom.xml [hunk 1]
```diff
@@ -1733,7 +1733,7 @@
               <configuration>
                 <rules>
                   <requireMavenVersion>
-                    <version>${maven.version}</version>
+                    <version>[3.3,)</version>
```
Comment (srowen) [inline pom.xml]: This should just be `[${maven.version},)` Because this minimum version is provided by the build it's OK to require it.

Comment (srowen) [inline pom.xml]: Actually, that's what the existing specification already means:
https://maven.apache.org/enforcer/enforcer-rules/versionRanges.html

Comment (jbonofre) [inline pom.xml]: ${maven.version} and [${maven.version},) are actually the same think for an enforcer perspective. So, if we don't go to [3.3,), then I will close this PR.

**Comment by srowen** [comment]
OK yes let's close this PR then @jbonofre

**Pull request closed without merging**